src/main/java/org/example/controller/handler/ResourceExceptionHandler.java

```diff
@@ -0,0 +1,23 @@
+package org.example.controller.handler;
+
+import org.example.exceptions.ResourceNotFoundException;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+import lombok.extern.slf4j.Slf4j;
+
+@ControllerAdvice
+@Slf4j
+public class ResourceExceptionHandler {
```

Comment:
you can use response code on an exception, it is simpler than having a separate exception handler
point 2 from https://www.baeldung.com/spring-response-status-exception
but not related to the topic of the course